tests/metricsSubmissionUrlTests.spec.ts

```diff
@@ -0,0 +1,90 @@
+import { getEndpointParams } from '../source/utils';
+
+describe('Test Submission URL information in metrics support', () => {
+  describe('Test universe name', () => {
+    it('Test submit.backtrace.io universe name', () => {
+      const expectedUniverseName = 'yolo';
+      const expectedToken =
+        '55555111112345eb7ae344f6e002de2e20c81fbdedf6991c2f3bb45b32b55555';
+      const testSubmissionUrl = `https://submit.backtrace.io/${expectedUniverseName}/${expectedToken}/json`;
+
+      const params = getEndpointParams(testSubmissionUrl);
+      if (!params) {
+        throw new Error(
+          `Invalid test case - the test submission URL ${testSubmissionUrl} is invalid for current test`,
+        );
+      }
+
+      expect(params.universe).toBe(expectedUniverseName);
+      expect(params.token).toBe(expectedToken);
+    });
+  });
+
+  it('Test invalid submit.backtrace.io universe name', () => {
+    const expectedUniverseName = 'yolo';
+    const expectedToken =
+      '55555111112345eb7ae344f6e002de2e20c81fbdedf6991c2f3bb45b32b55555';
+    const testSubmissionUrl = `https://submit.backtrace.io/${expectedUniverseName}/definitely-not-a-valid-url/${expectedToken}/json`;
+
+    const params = getEndpointParams(testSubmissionUrl);
+
+    expect(params).toBeUndefined();
+  });
+
+  it('Test invalid submit.backtrace.io without token', () => {
+    const expectedUniverseName = 'yolo';
+    const testSubmissionUrl = `https://submit.backtrace.io/${expectedUniverseName}`;
+
+    const params = getEndpointParams(testSubmissionUrl);
+
+    expect(params).toBeUndefined();
+  });
+
+  it('Test valid on-premise <universe>.backtrace.io url with configuration token', () => {
+    const expectedUniverseName = 'yolo';
+    const expectedToken = '55555111112345eb7ae344f6e002de2e20c81fbdedf6991c2f3bb45b32b55555'
+    const testSubmissionUrl = `https://${expectedUniverseName}.backtrace.io/`;
+
+    const params = getEndpointParams(testSubmissionUrl, expectedToken);
+    if (!params) {
+      throw new Error(
+        `Invalid test case - the test submission URL and ${testSubmissionUrl} and token ${expectedToken} are invalid for current test`,
+      );
+    }
+
+    expect(params.universe).toBe(expectedUniverseName);
+    expect(params.token).toBe(expectedToken);
+  });
+
+  it('Test valid on-premise <universe>.sp.backtrace.io url with configuration token', () => {
```

Comment:
shouldn't we also validate a validation for an on-premise URL with the token?

Reply:
`https://on_premise_url/<token>` isn't supported, if on-premise is used we require the token to be passed in the client configurations.
Should we add support for token in the url? I don't know if it makes sense, since the on-premise submissionURL doesn't have a /token in the path.

Reply:
no you're right. I messed up. THanks!